### ISSUE TITLE
Clean Up Proxy Minion Debug Logs

### ISSUE THREAD

**Issue by rallytime** [saltstack/salt#29342]
When running a proxy minion, the debug logs are pretty verbose for errors that don't pertain to the proxy minion that is running. For example, if I am running a proxy minion that is not an fx2 proxy minion I see the following in the debug logs anyway:
```
[CRITICAL] Failed to load grains defined in grain file fx2.fx2 in function <function fx2 at 0x7efd113370c8>, error:
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/loader.py", line 634, in grains
    ret = fun()
  File "/root/SaltStack/salt/salt/grains/fx2.py", line 83, in fx2
    return _grains()
  File "/root/SaltStack/salt/salt/grains/fx2.py", line 65, in _grains
    (username, password) = _find_credentials()
  File "/root/SaltStack/salt/salt/grains/fx2.py", line 47, in _find_credentials
    admin_password=p)
  File "/root/SaltStack/salt/salt/modules/dracr.py", line 1087, in get_chassis_name
    admin_password=admin_password)
  File "/root/SaltStack/salt/salt/modules/dracr.py", line 1317, in bare_rac_cmd
    admin_password=admin_password)
  File "/root/SaltStack/salt/salt/modules/dracr.py", line 124, in __execute_ret
    modswitch))
  File "/root/SaltStack/salt/salt/modules/cmdmod.py", line 611, in _run_all_quiet
    pillar_override=pillar_override)
  File "/root/SaltStack/salt/salt/modules/cmdmod.py", line 436, in _run
    .format(cmd, kwargs, exc)
CommandExecutionError: Unable to run command ['racadm', '-r', '10.4.28.150', '-u', 'root', '-p', 'saltstack', 'getchassisname'] with the context {'with_communicate': True, 'shell': False, 'env': {'LANG': 'en_US.UTF-8', 'TERM': 'xterm-256color', 'SHELL': '/bin/bash', 'LESSCLOSE': '/usr/bin/lesspipe %s %s', 'XDG_RUNTIME_DIR': '/run/user/1000', 'MAIL': '/var/mail/root', 'SHLVL': '1', 'PYTHONPATH': '/root/SaltStack/salt/', 'LESSOPEN': '| /usr/bin/lesspipe %s', 'PWD': '/root', 'LOGNAME': 'root', 'USER': 'root', 'PATH': '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/root/SaltStack/salt/scripts/', 'HOME': '/root', 'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arj=01;31:*.taz=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lz=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.axv=01;35:*.anx=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.axa=00;36:*.oga=00;36:*.spx=00;36:*.xspf=00;36:', 'XDG_SESSION_ID': '3', 'LC_ALL': 'C', '_': '/root/SaltStack/salt/scripts/salt-proxy'}, 'stdout': -1, 'close_fds': True, 'stdin': None, 'stderr': -1, 'cwd': '/root'}, reason: [Errno 2] No such file or directory
```
If you're running a different proxy minion, these errors are not helpful and make the debug logs needlessly muddy. 

The same thing goes for the junos proxy minion:
```
[CRITICAL] Failed to load grains defined in grain file junos.facts in function <function facts at 0x7efd11337488>, error:
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/loader.py", line 634, in grains
    ret = fun()
  File "/root/SaltStack/salt/salt/grains/junos.py", line 46, in facts
    if 'junos.facts' in __proxy__:
NameError: global name '__proxy__' is not defined
```

And the marathon proxy minion:
```
[CRITICAL] Failed to load grains defined in grain file marathon.marathon in function <function marathon at 0x7efd10ae6d70>, error:
Traceback (most recent call last):
  File "/root/SaltStack/salt/salt/loader.py", line 634, in grains
    ret = fun()
  File "/root/SaltStack/salt/salt/grains/marathon.py", line 46, in marathon
    decode=True,
  File "/root/SaltStack/salt/salt/utils/http.py", line 438, in query
    **req_kwargs
  File "/usr/lib/python2.7/dist-packages/tornado/httpclient.py", line 102, in fetch
    self._async_client.fetch, request, **kwargs))
  File "/usr/lib/python2.7/dist-packages/tornado/ioloop.py", line 445, in run_sync
    return future_cell[0].result()
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 215, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 3, in raise_exc_info
gaierror: [Errno -2] Name or service not known
```

FYI @cro

**Comment by cro** [comment]
This is somewhat related to #28886 -- most of those are coming from the grains modules which are not respecting the `__proxyenabled__` directives right now.

**Comment by rallytime** [comment]
That makes sense. I figured if it was something simple like making sure the `__virtual__` functions weren't loading inadvertently that would be a good place to look too, but I am just speculating.

**Comment by cro** [comment]
Fixed by #30293 and #30289

**Comment by rallytime** [comment]
This has been fixed. Thanks @cro!
